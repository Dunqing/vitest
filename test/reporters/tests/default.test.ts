import path from 'pathe'
import { describe, expect, test } from 'vitest'
import { runVitestCli } from '../../test-utils'

const resolve = (id = '') => path.resolve(__dirname, '../fixtures/default', id)
async function run(fileFilter: string[], watch = false, ...args: string[]) {
  return runVitestCli(
    ...fileFilter,
    '--root',
    resolve(),
    watch ? '--watch' : '--run',
    ...args,
  )
}

describe('default reporter', async () => {
  test('normal', async () => {
    const { stdout } = await run(['b1.test.ts', 'b2.test.ts'])
    expect(stdout).contain('✓ b2 test')
    expect(stdout).not.contain('✓ nested b1 test')
    expect(stdout).contain('× b failed test')
  })

  test('show full test suite when only one file', async () => {
    const { stdout } = await run(['a.test.ts'])
    expect(stdout).contain('✓ a1 test')
    expect(stdout).contain('✓ nested a3 test')
    expect(stdout).contain('× a failed test')
    expect(stdout).contain('nested a failed 1 test')
  })

  test('rerun should undo', async () => {
    const vitest = await run([], true, '-t', 'passed')

    // one file
    vitest.write('p')
    await vitest.waitForStdout('Input filename pattern')
    vitest.write('a\n')
    await vitest.waitForStdout('Filename pattern: a')
    await vitest.waitForStdout('Waiting for file changes')
    expect(vitest.stdout).contain('✓ a1 test')
    expect(vitest.stdout).contain('✓ nested a3 test')

    // rerun and two files
    vitest.write('p')
    await vitest.waitForStdout('Input filename pattern')
    vitest.write('b\n')
    await vitest.waitForStdout('Waiting for file changes')
    await vitest.waitForStdout('Filename pattern: b')
    expect(
      vitest.stdout
        .replace(/Start\sat.+\n?/, '')
        .replace(/Duration.+\n?/, ''),
    ).toMatchInlineSnapshot(`
      "? Input filename pattern › b✔ Input filename pattern … b

       RERUN  ../../change filename pattern
              Filename pattern: b
              Test name pattern: /passed/

       ✓ b1.test.ts (13)
       ✓ b2.test.ts (13)

       Test Files  2 passed (2)
            Tests  12 passed | 14 skipped (26)
            

       PASS  Waiting for file changes...
             press h to show help, press q to quit
      "
    `)
    expect(vitest.stdout).toContain('RERUN')
    expect(vitest.stdout).toContain('b1.test.ts')
    expect(vitest.stdout).toContain('b2.test.ts')
    expect(vitest.stdout).not.toContain('nested b1 test')
    expect(vitest.stdout).not.toContain('b1 test')
    expect(vitest.stdout).not.toContain('b2 test')
  })
}, 120000)
