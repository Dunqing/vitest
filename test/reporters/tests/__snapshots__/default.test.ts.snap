// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`default reporter > normal 1`] = `
"
 · default/b1.test.ts (13)
 · default/b2.test.ts (13)
 ❯ default/b1.test.ts (13)
   ✓ b1 passed (6)
   ❯ b1 failed (7)
     ✓ b1 test
     ✓ b2 test
     ✓ b3 test
     × b failed test
     ✓ nested b (3)
 ❯ default/b2.test.ts (13)
   ✓ b2 passed (6)
   ❯ b2 failed (7)
     ✓ b1 test
     ✓ b2 test
     ✓ b3 test
     × b failed test
     ✓ nested b (3)

 Test Files  2 failed (2)
      Tests  2 failed | 24 passed (26)"
`;

exports[`default reporter > show full test suite when only one file 1`] = `
"
 ❯ default/a.test.ts (13)
   ✓ a passed (6)
     ✓ a1 test
     ✓ a2 test
     ✓ a3 test
     ✓ nested a (3)
       ✓ nested a1 test
       ✓ nested a2 test
       ✓ nested a3 test
   ❯ a failed (7)
     ✓ a1 test
     ✓ a2 test
     ✓ a3 test
     × a failed test
     ✓ nested a (3)
       ✓ nested a1 test
       ✓ nested a2 test
       ✓ nested a3 test
 ❯ default/a.test.ts (13)
   ✓ a passed (6)
     ✓ a1 test
     ✓ a2 test
     ✓ a3 test
     ✓ nested a (3)
       ✓ nested a1 test
       ✓ nested a2 test
       ✓ nested a3 test
   ❯ a failed (7)
     ✓ a1 test
     ✓ a2 test
     ✓ a3 test
     × a failed test
     ✓ nested a (3)
       ✓ nested a1 test
       ✓ nested a2 test
       ✓ nested a3 test

 Test Files  1 failed (1)
      Tests  1 failed | 12 passed (13)"
`;
